src/utils/WindowDimensionsProvider.js

```diff
@@ -0,0 +1,35 @@
+/* eslint-disable no-undef */
+import React, { createContext, useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+const WindowDimensionsCtx = createContext(null);
+
+const WindowDimensionsProvider = ({ children }) => {
+  const [dimensions, setDimensions] = useState({
+    width: window.innerWidth,
+    height: window.innerHeight,
+  });
+
+  useEffect(() => {
+    const handleResize = () => {
+      setDimensions({
+        width: window.innerWidth,
+        height: window.innerHeight,
+      });
+    };
+    window.addEventListener('resize', handleResize);
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, []);
+
+  return <WindowDimensionsCtx.Provider value={dimensions}>{children}</WindowDimensionsCtx.Provider>;
+};
+
+WindowDimensionsProvider.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
```

Comment:
I imagine this is auto-generated but, What is this for exactly? The comment about disabling the eslint.

Reply:
This is done to ignore an es-lint rule on the next line or the whole document, like at the top. In this case, it was a mistake since it was copied from another file, it will be removed in the next pr.

You may see it in other files since proptypes is required by eslint but certain props need to be defined in a more specific way, for example, arrays, it needs to be arrayOf and contain the specifics of the array's shape and the same goes for objects.

The one at the top is to ignore the undefined "window" since there is no need to define it in React when we are rendering HTML but eslint will still complain about it.

Reply:
For sake of speed, going to resolve but I'm not sure we have to leave "for a future" PR the changes we have and can change right know with a little commit. Although thanks for the super complete explanation(: I didn't have any idea of these.